samples: dac: simplify meta data

hakehuang opened the pull request:
use config from dts to get the supporting status.
remove the platfrom_allow list.

martinjaeger commented on samples/drivers/dac/Kconfig:
```diff
@@ -0,0 +1,17 @@
+ZEPHYR_USER := zephyr,user
+
+config DAC_SAMPLE_RUN
+	bool "Run the application this many seconds"
```
Description doesn't look right.

martinjaeger reviewed (commented):
Nice improvement!

kartben commented on samples/drivers/dac/Kconfig:
```diff
@@ -0,0 +1,17 @@
+	bool "Run the application this many seconds"
+	default y if $(dt_node_has_prop,/$(ZEPHYR_USER),dac)
+	help
+	  platfrom supports dac sample.
```
please address typo

hakehuang committed:
samples: dac: simplify meta data

use config from dts to get the supporting status.
remove the platfrom_allow list.

Signed-off-by: Hake Huang <hake.huang@nxp.com>

kartben reviewed (dismissed):
this really looks like a "hack", not sure it's desirable? @nashif can maybe chime in

hakehuang commented:
> this really looks like a "hack", not sure it's desirable? @nashif can maybe chime in

tests/drivers/dac use similar way, with this, we can remove the `platform_allow` constrains totally.

erwango reviewed (commented):
I won't block, as indeed some improvements are required for dac samples, but this isn't sufficient to me. One required improvement would be:
- add a `dac0` alias in all boards supporting dac (allows to remove `dac` prop of `zephyr,user`)
- For `dac-channel-id` and `dac-resolution` use an app.overlay which fits to most of the boards and rely on boards/<foo_board>.overlay when not working.

hakehuang commented:
> I won't block, as indeed some improvements are required for dac samples, but this isn't sufficient to me. One required improvement would be:
> 
> * add a `dac0` alias in all boards supporting dac (allows to remove `dac` prop of `zephyr,user`)
> * For `dac-channel-id` and `dac-resolution` use an app.overlay which fits to most of the boards and rely on boards/<foo_board>.overlay when not working.

I agree? shall I update in this PR? @erwango , @nashif , @kartben ?

erwango commented:
> I agree. shall I update in this PR? @erwango , @nashif , @kartben ?

I think it is worth having a look now, while you're at.
